frameworks/keyed/react-hooks/src/main.jsx

```diff
@@ -34,35 +39,47 @@ function listReducer(state, action) {
       return { data: data.concat(buildData(1000)), selected };
     case 'UPDATE':
       const newData = data.slice(0);
+
       for (let i = 0; i < newData.length; i += 10) {
         const r = newData[i];
+
         newData[i] = { id: r.id, label: r.label + " !!!" };
       }
       return { data: newData, selected };
     case 'CLEAR':
       return { data: [], selected: 0 };
     case 'SWAP_ROWS':
-      debugger;
-      return (data.length > 998) ? { data: [data[0], data[998], ...data.slice(2, 998), data[1], data[999]] } : state;
+      return { data: [data[0], data[998], ...data.slice(2, 998), data[1], data[999]] };
```

Comment:
Probably shouldn't remove the length check here. While unnecessary for the test runs it is necessary for the operation not to die horribly after a couple of rows have been removed.